<!doctype html>
<html>
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=Edge,chrome=1">
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
	<meta charset="utf-8">
	
		<title>iOS simulator close because not valid the try {} catch() » Community Questions &amp; Answers » Appcelerator Developer Center</title>
	
	<meta name="viewport" content="width=device-width,initial-scale=1">
	<link href="../../related/favicon.png" rel="shortcut icon" type="image/png">
	<link href="//fonts.googleapis.com/css?family=Open+Sans:300,400,500,600,700" rel="stylesheet" type="text/css">
	<link href="//d3ilu1xuwhtfe2.cloudfront.net/e8e0ebd/css/style.css" rel="stylesheet" type="text/css">
	<link href="../../related/qa.css" rel="stylesheet" type="text/css">
	<link href="../../related/hybrid.css" rel="stylesheet" type="text/css">
	
	<link href="../../related/overrides.css" rel="stylesheet" type="text/css">
</head>
<body>

<div class='mast'>    <div class='container'>        <div class='row'>            <div class='col-sm-12'>                <div class='masthead'><h1>Titanium Community Questions & Answer Archive</h1><h2>We felt that 6+ years of knowledge should not die so this is the Titanium Community Questions &amp; Answer Archive</h2></div>            </div>        </div>    </div></div>

<main class="container">
	<div class="row">
		<div class="col-sm-12">
			
		</div>
	</div>
	<div class="row">
		<div class="col-sm-12">
			<article class="question-container">
	<div class="question">
		<header>
			<h1>iOS simulator close because not valid the try {} catch()</h1>
		</header>
		<section>
			<p>I have this try in my code:</p>
<p>`try{</p>
<pre><code class="hljs">    var checkPedidoColumn = db.execute('<span class="hljs-operator"><span class="hljs-keyword">ALTER</span> <span class="hljs-keyword">TABLE</span> pedido_item <span class="hljs-keyword">ADD</span> <span class="hljs-keyword">COLUMN</span> <span class="hljs-string">"min_tamanho"</span> <span class="hljs-built_in">NUMERIC</span><span class="hljs-string">');

} catch(e) {

    Ti.API.info('</span>the <span class="hljs-keyword">column</span> min_tamanho <span class="hljs-keyword">exists</span> <span class="hljs-keyword">in</span> the <span class="hljs-keyword">table</span> pedido_item<span class="hljs-string">');
}`</span></span>
</code></pre><p>This table and column exists and try this ok, until the iOS 8.3.<br>Now the titanium return:</p>
<p>A SQLite database error occurred on database &#39;&#x2F;Users&#x2F;maicocardoso&#x2F;Library&#x2F;Developer&#x2F;CoreSimulator&#x2F;Devices&#x2F;B7AFD1AF-767E-42CD-A370-C2D904417918&#x2F;data&#x2F;Containers&#x2F;Data&#x2F;Application&#x2F;C1F57F5A-311C-48A1-8042-1C1FEEC20DED&#x2F;Library&#x2F;Private Documents&#x2F;database3.sql&#39;: Error Domain=com.plausiblelabs.pldatabase Code=3 &quot;An error occured parsing the provided SQL statement.&quot; UserInfo=0x797aea90 {com.plausiblelabs.pldatabase.error.vendor.code=1, NSLocalizedDescription=An error occured parsing the provided SQL statement., com.plausiblelabs.pldatabase.error.query.string=ALTER TABLE pedido_item ADD COLUMN &quot;min_tamanho&quot; NUMERIC, com.plausiblelabs.pldatabase.error.vendor.string=duplicate column name: min_tamanho} (SQLite #1: duplicate column name: min_tamanho) (query: &#39;ALTER TABLE pedido_item ADD COLUMN &quot;min_tamanho&quot; NUMERIC&#39;)</p>
<p>And the simulator close.I would like to now if the rule change or this is a bug. thanks!</p>

		</section>
		<footer>
			<div class="author">
				&mdash; asked <span title="April 14th 2015, 10:43:09 pm">April 14th 2015</span>
				by <span class='authorname'>Maico Cardoso</span>
			</div>

			
				<ul class="tags">
					
						<li><span class='tagname'>catch</span></li>
					
						<li><span class='tagname'>ios</span></li>
					
						<li><span class='tagname'>ipad</span></li>
					
						<li><span class='tagname'>sqlite</span></li>
					
						<li><span class='tagname'>try</span></li>
					
				</ul>
			

			<section>
				<h5>2 Comments</h5>
				
					<ul class="comments">
					
						<li class="comment">
							<div><p>I have an similar issue with JSON.Parse, when I try to use JSON.Parse<br>with an invalid JSON the simulator closes and ignoring the surrounded tr&#x2F;catch block.</p>
<p>I surrounded JSON.Parse with an regex which ensures that the json is valid.</p>
<p>I would suggest to go an similar way,<br>check if column already exists <strong>PRAGMA table_info(table-name);</strong><br>and depending on this decide whether to add a new column or not.</p>
<p>I guess that&#39;s a bug, but I don&#39;t care about this…<br>Generally it isn&#39;t a good design to use error handling for your<br>program flow.</p>
<p>If it&#39;s a bug or not, the better solution to check if the column exists<br>and act depending on this decide what to do.</p>
</div>
							<div class="author">
								&mdash; commented <span title="April 15th 2015, 1:04:14 pm">April 15th 2015</span>
								by <span class='authorname'>Michael Bahl</span>
							</div>
						</li>
					
						<li class="comment">
							<div><p>Hi Jan. Are you right, i try do it and it&#39;s ok. Now i am creating a query to check firts. i would like that the rule turn back, but problably will not.<br>Thanks.</p>
</div>
							<div class="author">
								&mdash; commented <span title="April 24th 2015, 2:45:33 pm">April 24th 2015</span>
								by <span class='authorname'>Maico Cardoso</span>
							</div>
						</li>
					
					</ul>
				
			</section>
		</footer>
		<aside class="vote-box">
			<div class="score"><span>0</span> Votes</div>
			<div class="answers"><span>5</span> Answers</div>
		</aside>
	</div>

	<hr>

	<h3>5 Answers</h3>

	
		<ul class="answers">
			
				<li class="answer">
				
					<article id="answer-287157">
				
						<section>
							<p>Hello,</p>
<p>I&#39;ve got the same issue while trying to insert existing items in plsqlite database :&#39;(<br>Was ok until iOs 8.3 for me too….</p>
<p>I think &quot;try catch&quot; isn&#39;t working as expected anymore …</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 24th 2015, 12:05:38 pm">April 24th 2015</span>
								by <span class='authorname'>Yoann WYFFELS</span><br>
								<a class="icon-bg icon-link" href="question/182008/ios-simulator-close-because-not-valid-the-try--catch#answer-287157" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-287158">
				
						<section>
							<p>Does anyone know if we - as regular users - can report issue&#x2F;bug to appceleator ?</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 24th 2015, 12:11:17 pm">April 24th 2015</span>
								by <span class='authorname'>Yoann WYFFELS</span><br>
								<a class="icon-bg icon-link" href="question/182008/ios-simulator-close-because-not-valid-the-try--catch#answer-287158" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-287193">
				
						<section>
							<p>I have to fall back to SDK 3.4.1 GA in order to use working &quot;try catch&quot;. This is clearly an issue in 3.5.x branch.</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 28th 2015, 6:18:54 am">April 28th 2015</span>
								by <span class='authorname'>Yoann WYFFELS</span><br>
								<a class="icon-bg icon-link" href="question/182008/ios-simulator-close-because-not-valid-the-try--catch#answer-287193" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-287195">
				
						<section>
							<p>I&#39;ve created issue in JIRA:<br><a href="https:&#x2F;&#x2F;jira.appcelerator.org&#x2F;browse&#x2F;AC-105">JIRA AC-105</a></p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="April 28th 2015, 6:28:09 am">April 28th 2015</span>
								by <span class='authorname'>Yoann WYFFELS</span><br>
								<a class="icon-bg icon-link" href="question/182008/ios-simulator-close-because-not-valid-the-try--catch#answer-287195" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
				<li class="answer">
				
					<article id="answer-287237">
				
						<section>
							<p>The problem is when building for iOS simulator iPad 2 and Retina if building for iOS iPad simulator Air or Resizable iPad does not occur any problem.</p>

						</section>
						<footer>
							<div class="author">
								&mdash; answered <span title="May 1st 2015, 11:15:29 am">May 1st 2015</span>
								by <span class='authorname'>Maico Cardoso</span><br>
								<a class="icon-bg icon-link" href="question/182008/ios-simulator-close-because-not-valid-the-try--catch#answer-287237" rel="permalink">permalink</a>
							</div>

							<h5>0 Comments</h5>
							
						</footer>
						<aside class="vote-box">
							<div class="score"><span>0</span> Votes</div>
						</aside>
					</article>
				</li>
			
		</ul>
	
</article>

		</div>
		<div class="col-sm-3"></div>
	</div>
	<div class="row"><div class="signoff">The ownership of individual contributions to this community generated content is retained by the authors of their contributions.<br>All trademarks remain the property of the respective owner.</div></div>
</main>


</body>
</html>
